executable-spec/src/Cardano/Ledger/Spec/State/ProposalsState.hs

```diff
@@ -2,38 +2,64 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeApplications #-}
 
 -- | Information about different proposals in the update system.
 --
 module Cardano.Ledger.Spec.State.ProposalsState
   ( ProposalsState
+  , ProposalsStateCompact
   , tally
+  , tallyCompact
   , revealProposal
+  , revealProposalCompact
   , updateBallot
+  , updateBallotCompact
   , votingPeriodStarted
   , votingPeriodEnded
   , votingPeriodHasNotEnded
   , decision
+  , decisionCompact
   , isRevealed
   , isNotRevealed
   )
 where
 
 import           Control.Exception (assert)
-import qualified Data.Map as Map
 import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
 
 import           Ledger.Core (BlockCount, Slot)
+import           Cardano.Binary (ToCBOR, toCBOR)
+import           Cardano.Crypto.DSIGN.Class (SignedDSIGN)
+import qualified Cardano.Crypto.DSIGN.Class as Crypto.DSIGN
+import           Cardano.Crypto.DSIGN.Mock (MockDSIGN)
+import           Cardano.Crypto.DSIGN.Mock (SignKeyDSIGN (SignKeyMockDSIGN),
+                     VerKeyDSIGN (VerKeyMockDSIGN))
+
 
 import           Cardano.Ledger.Spec.Classes.Hashable (HasHash, Hash, Hashable,
                      hash)
 import           Cardano.Ledger.Spec.Classes.HasSigningScheme (VKey)
 import           Cardano.Ledger.Spec.State.ProposalState (Decision,
-                     HasVotingPeriod, IsVote, ProposalState, VotingPeriod,
-                     newProposalState)
+                     HasVotingPeriod, IsVote, ProposalState, VotingPeriod, ProposalStateCompact,
+                     newProposalState, newProposalStateCompact)
 import qualified Cardano.Ledger.Spec.State.ProposalState as ProposalState
-import           Cardano.Ledger.Spec.State.StakeDistribution (StakeDistribution)
+import           Cardano.Ledger.Spec.State.StakeDistribution (StakeDistribution, StakeDistributionCompact)
+import           Cardano.Ledger.Spec.STS.Common.Compact (CompactHash, toCompactHash)
+import           Cardano.Ledger.Spec.STS.Common.Crypto (BenchCrypto)
+
+
+newtype ProposalsStateCompact = ProposalsStateCompact (Map CompactHash ProposalStateCompact )
```

Comment:
We shouldn't need to change this for using compact hashes. The `Hash` instance would take care of this.